pkg/controllers/namespace/namespaceController.go

```diff
@@ -0,0 +1,285 @@
+package namespace
+
+import (
+	"reflect"
+	glog "github.com/Sirupsen/logrus"
+	calicocache "github.com/projectcalico/k8s-policy/pkg/cache"
+	"github.com/projectcalico/k8s-policy/pkg/controllers/controller"
+	"github.com/projectcalico/k8s-policy/pkg/converter"
+	"github.com/projectcalico/libcalico-go/lib/api"
+	"github.com/projectcalico/libcalico-go/lib/client"
+	"github.com/projectcalico/libcalico-go/lib/errors"
+	"k8s.io/apimachinery/pkg/fields"
+	uruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api/v1"
+	rest "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"strings"
+	"time"
+)
+
+// NamespaceController Implements Controller interface
+// Responsible for monitoring kubernetes namespaces and
+// syncing them to Calico datastore.
+type NamespaceController struct {
+	indexer        cache.Indexer
+	informer       cache.Controller
+	calicoObjCache calicocache.ResourceCache
+	calicoClient   *client.Client
+}
+
+// NewNamespaceController Constructor for NamespaceController
+func NewNamespaceController(k8sConfig *rest.Config) controller.Controller {
+
+	// Get kubenetes clientset
+	clientset, err := kubernetes.NewForConfig(k8sConfig)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	cconfig, err := client.LoadClientConfig("")
+	if err != nil {
+		panic(err)
+	}
+
+	// Get Calico client
+	cclient, err := client.New(*cconfig)
+	if err != nil {
+		panic(err)
+	}
+
+	// Function returns slice of profile objects stored by policy controller
+	// on ETCD datastore.Indentifies controller writen objects by
+	// their naming convention.
+	listFunc := func() ([]interface{}, error) {
+
+		var filteredProfiles []interface{}
+
+		// Get all profile objects from ETCD datastore
+		calicoProfiles, err := cclient.Profiles().List(api.ProfileMetadata{})
+		if err != nil {
+			return filteredProfiles, err
+		}
+
+		// Filter out only objects that are written by policy controller
+		for _, profile := range calicoProfiles.Items {
+
+			profileName := profile.Metadata.Name
+			if strings.HasPrefix(profileName, converter.ProfileNameFormat) {
+				filteredProfiles = append(filteredProfiles, profile)
+			}
+		}
+		glog.Debugf("Found %d profiles in calico ETCD:", len(filteredProfiles))
+		return filteredProfiles, nil
+	}
+
+	// Function returns key of the object in kubernetes format.
+	keyFunc := func(obj interface{}) string {
+		profile := obj.(api.Profile)
+		Key := profile.Metadata.Name
```

Comment:
I've been thinking about this more - I think KeyFunc needs to return the k8s representation of the key (i.e. `namespace.name`).

If we store Calico object names in the cache, then we don't have the flexibility to change how Calico names its objects in the future (which we may want to do).